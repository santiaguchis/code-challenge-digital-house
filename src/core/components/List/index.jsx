import PropTypes from 'prop-types';	
import { ScrollView, View } from 'react-native';

const List = ({
  children
}) => {
  return (
    <View>
      <ScrollView>
        {children}
      </ScrollView>
    </View>
  )
}

List.defaultProps = {
  items: [],
}

List.propTypes = {
  items: PropTypes.array,
}

export {
  List
}