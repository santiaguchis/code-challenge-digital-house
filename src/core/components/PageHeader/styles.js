import { StyleSheet } from 'react-native';

import { theme } from 'src/core/utils/styles';

const styles = StyleSheet.create({
  pageHeader: {
    backgroundColor: theme.colors.primaryLight,
    height: 150,
    justifyContent: 'flex-end',
    padding: 20,
  },
  pageTitle: {
    fontSize: 24,
    fontWeight: '800',
    lineHeight: 24,
  }
});

export default styles;