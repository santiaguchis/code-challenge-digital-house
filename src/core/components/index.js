export * from './Button';
export * from './Card';
export * from './List';
export * from './ListHeader';
export * from './ListItem';
export * from './Page';
export * from './PageContent';
export * from './PageFooter';
export * from './PageHeader';
export * from './Separator';
export * from './Space';