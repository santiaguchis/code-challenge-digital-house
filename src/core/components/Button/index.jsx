import PropTypes from 'prop-types';
import { Text, TouchableOpacity } from 'react-native'

import styles from './styles';

const {
  button,
  text,
} = styles;

const Button = ({
  size,
  style,
  onPress,
  children,
}) => {
  const goTo = () => {
    onPress()
  }
  const styleText = {
    ...text,
    fontSize: size === 'sm' ? 12 : 16
  }
  return (
    <TouchableOpacity
      onPress={ goTo }
      style={[button,style]}
    >
      <Text 
        testID='text-button' 
        style={ styleText }
      >{children}</Text>
    </TouchableOpacity>
  )
}

Button.defaultProps = {
  style: {},
  size: 'sm',
  onPress: () => {},
}

Button.propTypes = {
  style: PropTypes.object,
  size: PropTypes.string,
  onPress: PropTypes.func,
  children: PropTypes.string.isRequired,
}

export {
  Button
}