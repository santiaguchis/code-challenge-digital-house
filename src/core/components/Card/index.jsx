import { View } from 'react-native';

import styles from './styles';

const {
  card
} = styles;

const Card = ({ style = {}, children }) => {
  return (
    <View style={ [card, style] }>
      { children }
    </View>
  )
}

export default Card;

export {
  Card
}