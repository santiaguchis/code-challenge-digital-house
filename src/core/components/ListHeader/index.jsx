import PropTypes from 'prop-types';

import { StyleSheet, Text, View } from 'react-native';

import styles from './styles';

const {
  container,
  title,
} = styles;

const ListHeader = ({ textToUpper, children }) => {
  const titleStyle = {
    ...title,
    textTransform: textToUpper ? 'uppercase' : 'none'
  }
  return (
    <View style={ container }>
      <Text style={ titleStyle }>
      { children }
      </Text>
    </View>
  )
}

ListHeader.propTypes = {
  textToUpper: false,
  children: '',
};

ListHeader.propTypes = {
  textToUpper: PropTypes.bool,
  children: PropTypes.string
};

export default ListHeader;

export {
  ListHeader
}