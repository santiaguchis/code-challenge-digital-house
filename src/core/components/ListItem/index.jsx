import PropTypes from 'prop-types';	
import { Text, TouchableOpacity, View } from 'react-native';

import styles from './styles';

const {
  action,
  actionIcon
} = styles

const ListItem = ({
  onPress,
  style,
  children,
}) => {
  return (
    <TouchableOpacity
      onPress={ onPress }
      style={ style }
    >
      { children }
      <View style={ action }>
        <Text style={ actionIcon }>&gt;</Text>
      </View>
    </TouchableOpacity>
  )
}

ListItem.defaultProps = {
  onPress: () => {},
  style: {},
}

ListItem.propTypes = {
  onPress: PropTypes.func,
  style: PropTypes.object,
  children: PropTypes.any
}

export default ListItem;

export {
  ListItem
}