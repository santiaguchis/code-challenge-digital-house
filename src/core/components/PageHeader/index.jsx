import { StatusBar, Text, View  } from 'react-native';

import { theme } from 'src/core/utils/styles';

import styles from './styles';

const {
  pageHeader,
  pageTitle
} = styles;

const PageHeader = ({
  title
}) => {
  return (
    <View style={ pageHeader }>
      <StatusBar
        backgroundColor={ theme.colors.primaryLight }
      />
      <Text style={ pageTitle }>{ title }</Text>
    </View>
  )
}

export {
  PageHeader
}