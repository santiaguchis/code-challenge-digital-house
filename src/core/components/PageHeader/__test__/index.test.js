import { render, screen } from '@testing-library/react-native';
import { PageHeader } from 'src/core/components/PageHeader';

let component;
describe('PageHeader core/component', () => {
  const placeholderText = 'Hello Page Header'
  beforeEach(() => {
    component = render(<PageHeader title={ placeholderText } />)
  })

  it('renderer element', () => {
    expect(component).toBeDefined();
  })
  it('Check text in component', () => {
    const queryText = screen.getByText(placeholderText);
    expect(queryText).toHaveTextContent(placeholderText);
  })

  it('Check snapshot', () => {
    expect(screen.toJSON()).toMatchSnapshot();
  })
})