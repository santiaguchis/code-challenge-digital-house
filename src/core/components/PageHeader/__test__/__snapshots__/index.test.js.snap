// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader core/component Check snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#CFD6FF",
      "height": 150,
      "justifyContent": "flex-end",
      "padding": 20,
    }
  }
>
  <Text
    style={
      {
        "fontSize": 24,
        "fontWeight": "800",
        "lineHeight": 24,
      }
    }
  >
    Hello Page Header
  </Text>
</View>
`;
